### ISSUE TITLE
suggestion: split redirector into multiple server types

### ISSUE THREAD

**Issue by tecknicaltom** [ezra-buckingham/terry-the-terraformer#55]
A slight suggestion that you can feel free to close and ignore, but after getting familiar with the Terry code, it looks like it could be cleaner if the redirector server type were split into separate server types (e.g. dns_redirector). There are several places with conditional logic based on `redirector_type` and there's not much shared logic between these server types.

Thanks again for this great project. If you're not opposed to this idea, I may be able to quickly do it and submit a PR, though I'm not sure how it would handle upgrading existing operations' servers.

**Comment by ezra-buckingham** [comment]
I love this idea too and fully agree with you. I had more so just kept going down the first path (having them be the same server type and then using that `redirector_type` be the logic that separates the two. I may have some time to refactor it, but if you want to take a stab at it, go for it!

**Comment by ezra-buckingham** [comment]
Need to add an `http` redirector too

**Comment by leebaird** [comment]
I agree, knowing what type of redirector would be very helpful (http, https, dns).
